### ISSUE TITLE
Installation integration issues with proxies

### ISSUE THREAD

**Issue by aep-sunlife** [composer/composer#12240]
When I follow the stated instructions for installing PHP in an environment that involves proxies, then the documented commands do not work.

https://getcomposer.org/download/

Specifically, the `copy()` interacts poorly with net nannies that require proxies in certain environments.

I can manually tweak that command to include a proxy ad hoc, by supplying a `stream_context_set_default()` argument. However, that's not very portable, because some installations operate in proxied environments, and others operate in unproxied environments.

What is the recommended way to configure PHP to permanently use a proxy for all `copy()` commands?

We can link to that snippet just before the first `copy()` command, so that the instructions remain highly portable, regardless of the exact proxied vs. unproxied environment.

**Comment by johnstevenson** [comment]
> What is the recommended way to configure PHP to permanently use a proxy for all copy() commands?

There isn't one, but a lot of tools recognize proxy environment variables, which you need when using Composer behind a proxy.  See https://getcomposer.org/doc/faqs/how-to-use-composer-behind-a-proxy.md

So you could use `curl` or `wget` to download and save the installer script:

```sh
curl -o composer-setup.php  https://getcomposer.org/installer

wget -O composer-setup.php https://getcomposer.org/installer
```

**Comment by Seldaek** [comment]
Yeah the problem is we do not know if you have curl or wget installed, hence why we do everything with php like this. I don't think we can really fix this.
